di-ipv-evcs-stub/lambdas/src/services/evcsService.ts

```diff
@@ -0,0 +1,158 @@
+
+import PostRequest from "../domain/postRequest";
+import ServiceResponse from "../domain/serviceResponse";
+import VcState from "../domain/enums/vcState";
+import EvcsVcItem from "../model/evcsVcItem";
+
+import { config } from "../common/config";
+import { getSsmParameter } from "../common/ssmParameter";
+import { v4 as uuid } from "uuid";
+import PatchRequest from "../domain/patchRequest";
+
+const dynamoClient = config.isLocalDev
+  ? new DynamoDB({
+      endpoint: config.localDynamoDbEndpoint,
+      region: config.region,
+    })
+  : new DynamoDB({ region: config.region });
+
+export async function processPostUserVCsRequest(
+  userId: string,
+  postRequest: PostRequest,
+): Promise<ServiceResponse> {
+  console.info(`Post user record.`);
+  for (const persistVC of postRequest.persistVCs) {
+    const vcItem: EvcsVcItem = {
+      userId: userId,
+      vc: persistVC.vc,
+      vcSignature: persistVC.vc.split(".")[2],
+      state: persistVC.state,
+      metadata: persistVC.metadata!,
+      provenance: persistVC.provenance!,
+      ttl: await getTtl(),
+    };
+    await saveUserVC(vcItem);
+  }
+
+  return {
+    response: {
+      messageId: uuid(),
+    },
+    statusCode: 202,
+  };
+}
+
+export async function processGetUserVCsRequest(
+  userId: string,
+): Promise<ServiceResponse> {
+  console.info(`Get user record.`);
+  const getItemInput: QueryInput = {
+    TableName: config.evcsStubUserVCsTableName,
+    KeyConditionExpression: "#userId = :userIdValue",
+    FilterExpression: "#state = :stateValue",
+    ExpressionAttributeNames: {
+      "#userId": "userId",
+      "#state": "state",
+    },
+    ExpressionAttributeValues: {
+      ":userIdValue": { S: userId },
+      ":stateValue": { S: VcState.CURRENT },
+    },
+  };
+  // console.info(`Query Input - ${JSON.stringify(getItemInput)}`);
+  const items = (await dynamoClient.query(getItemInput)).Items ?? [];
+  console.info(`Total VCs retrived - ${items?.length}`);
+  const vcItems = items
+    .map((item) => unmarshall(item) as EvcsVcItem)
+    .map((evcsItem) => ({
+      vc: evcsItem.vc,
+      state: evcsItem.state as VcState,
+      metadata: evcsItem.metadata,
+    }));
+
+  return {
+    response: {
+      vcs: vcItems,
+      afterKey: "pagination will be implemented later",
+    },
+    statusCode: 200,
+  };
+}
+
+export async function processPatchUserVCsRequest(
+  userId: string,
+  postRequest: PatchRequest,
+): Promise<ServiceResponse> {
+  console.info(`Patch user record.`);
+  for (const updateVC of postRequest.updateVCs) {
+    const vcItem: EvcsVcItem = {
+      userId: userId,
+      vcSignature: updateVC.signature,
+      state: updateVC.state,
+      metadata: updateVC.metadata!,
+      ttl: await getTtl(),
+    };
+    await updateUserVC(vcItem);
+  }
+
+  return {
+    response: {
+      messageId: uuid(),
+    },
+    statusCode: 204,
+  };
+}
+
+async function saveUserVC(evcsVcItem: EvcsVcItem) {
+  console.info(`Save user vc.`);
```

Comment:
Do we have anything like session ID that we can use for all of these logs to give them a bit more context? Might be a bit too convoluted to implement if we don't already have it, but it would definitely make these easier to follow

Reply:
Only thing we got is userId .....which we can't add in log.